courses/management/commands/manage_program_certificates.py

```diff
@@ -0,0 +1,164 @@
+
+2. Forcefully Generate Program certificate for a user use -f or --force
+./manage.py manage_program_certificates —-create -—program=<program_readable_id> -—user=<username or email> -f
+
+**Revoke/Un-revoke Program Certificates**
+
+3. Revoke a program certificate for a user
+./mange.py manage_program_certificates -—revoke -—user=<username or email> -—program=<program_readable_id>
+
+4. Un-Revoke a program certificate for a user
+./mange.py manage_program_certificates -—unrevoke —-program=<program_readable_id> -—user=<username or email>
+
+"""
+
+from django.core.management.base import BaseCommand, CommandError
+from users.api import fetch_user
+from courses.api import manage_program_certificate_access, generate_program_certificate
+from courses.models import Program
+from django.contrib.auth import get_user_model
+
+User = get_user_model()
+
+
+class Command(BaseCommand):
+    """
+    Invoke with:
+
+        python manage.py manage_program_certificates
+    """
+
+    help = "Revoke, un revoke or create a program certificate for a program for the given User"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--user",
+            type=str,
+            help="The id, email or username of the enrolled User",
+            required=True,
+        )
+        group = parser.add_mutually_exclusive_group()
+        group.add_argument(
+            "--program", type=str, help="The 'readable_id' value for a Program"
+        )
+        parser.add_argument(
+            "--revoke", dest="revoke", action="store_true", required=False
+        )
+        parser.add_argument(
+            "--unrevoke", dest="unrevoke", action="store_true", required=False
+        )
+        parser.add_argument(
+            "--create", dest="create", action="store_true", required=False
+        )
+        parser.add_argument(
+            "-f",
+            "--force",
+            action="store_true",
+            dest="force",
+            help=(
+                "If provided, the certificate will be generated even if all required "
+                "courses for the program are not passed"
+            ),
+        )
+
+        super().add_arguments(parser)
+
+    def handle(self, *args, **options):  # pylint: disable=too-many-locals
+        """Handle command execution"""
+
+        revoke = options.get("revoke")
+        unrevoke = options.get("unrevoke")
+        create = options.get("create")
+        program = options.get("program")
+        user = options.get("user")
+        force_create = options.get("force", False)
+
+        if not (revoke or unrevoke) and not create and not user:
+            raise CommandError(
+                "The command needs a valid action e.g. --revoke, --unrevoke, --create."
+            )
+
+        # A user is needed for revoke/un-revoke and certificate creation
+        if not user:
+            raise CommandError("The command needs a valid user.")
+
```

Comment:
I've removed this check but this causes a test failure because we can't pass None as an argument to the `fetch_user()` method. So, I removed the `None` as the `user` parameter from `test_program_certificate_management_revoke_unrevoke_invalid_args`